.github/workflows/publish.yaml

```diff
@@ -45,3 +47,34 @@ jobs:
             oci-archive:"${rock_file}" \
             docker-daemon:"ghcr.io/canonical/${image_name}:${version}"
           docker push ghcr.io/canonical/${image_name}:${version}
+
+  oci-factory:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4
+      - name: Golang setup
+        uses: actions/setup-go@v5
+        with:
+          go-version: '>=1.22.0'
+      # install oci-factory via golang and set path in environment
+      - name: Install oci-factory
+        run: |
+          sudo apt update && sudo apt install -y git
+          go install github.com/canonical/oci-factory/tools/cli-client/cmd/oci-factory@latest
+          go install github.com/mikefarah/yq/v4@v4.44.3
+          echo "OCI_FACTORY=$(go env GOPATH)/bin/oci-factory" >> $GITHUB_ENV
+          echo "YQ=$(go env GOPATH)/bin/yq" >> $GITHUB_ENV
+        env:
+          GO111MODULE: "on"
+      - name: Set EOLs and version
+        # special case for hydra due to the canonical fork  
+        run: |
+          echo EOL_STABLE=$(date -d "$(date +'%Y-%m-%d') +3 month" "+%Y-%m-%d") >> $GITHUB_ENV
+          echo IMAGE_VERSION_STABLE=$($YQ '.version | split(".").0' rockcraft.yaml) >> $GITHUB_ENV
```

Comment:
plan is to support `hydra` `v2` and not care about minor versions b but keep pushing

happy to add minors but we should skip it for the canonical fork

Reply:
My 2 cents: for hydra which is so crucial for our platform and admin-ui, I'd care about minor versions too, but I'd like to know what the rest of the team thinks

Reply:
idea is we provide maintenance and cover cve patches, so using only the majority version allows us to upgrade continuously to a minor and stick with that tag